src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/DatePartitionedRecordsWriterFactory.java

```diff
@@ -128,6 +124,6 @@ private LocalDate getCurrentDate() {
   }
 
   private String constructFileName() {
-    return "dwhassessment_" + loggerId + "_" + logFileCount + ".avro";
+    return "dwhassessment_" + clock.instant().toEpochMilli() + "_" + loggerId + ".avro";
```

Comment:
I would suggest to use the format, e.g. `YYYY-MM-DDTHHMMSS.SSS` (colons omitted, because they are not a good filename character due to them being a separator in unix environments). Still allows sorting and it is human-readable.